File: solarsystem/src/main/java/fail/stderr/usb/kerbol2/KerbolSystem.java
package fail.stderr.usb.kerbol2;

import org.orekit.bodies.CelestialBody;
import org.orekit.frames.Frame;
import org.orekit.orbits.KeplerianOrbit;

public class KerbolSystem {

  public final Frame kerbolFrame = KerbolBodies.createKerbolFrame();

  public final KeplerianOrbit kerbinOrbit = KerbolBodies.createKerbinOrbit(kerbolFrame);
  public final Frame kerbinFrame = KerbolBodies.createKerbinFrame(kerbolFrame, kerbinOrbit);
  public final CelestialBody kerbin = KerbolBodies.createKerbin(kerbinFrame, kerbinOrbit);

  public final KeplerianOrbit munOrbit = KerbolBodies.createMunOrbit(kerbinFrame);
  public final Frame munFrame = KerbolBodies.createMunFrame(kerbinFrame, munOrbit);
  public final CelestialBody mun = KerbolBodies.createMun(munFrame, munOrbit);

  public final KeplerianOrbit minmusOrbit = KerbolBodies.createMinmusOrbit(kerbinFrame);
  public final Frame minmusFrame = KerbolBodies.createMinmusFrame(kerbinFrame, minmusOrbit);
  public final CelestialBody minmus = KerbolBodies.createMinmus(minmusFrame, minmusOrbit);

}

File: solarsystem/src/main/kotlin/fail/stderr/usb/kerbol/KerbolSystem.kt
package fail.stderr.usb.kerbol

import fail.stderr.usb.common.CustomCelestialBody
import org.orekit.time.AbsoluteDate
import org.orekit.time.TimeScale
import org.orekit.time.createConstantOffsetTimeScale

class KerbolSystem {

  companion object {

    val TIME_SCALE: TimeScale = createConstantOffsetTimeScale("KERBOLTIME", 0.0)

    val REF_DATE = AbsoluteDate(1983, 7, 14, TIME_SCALE)

  }

  val kerbolFrame = createKerbolFrame()

  val kerbinOrbit = createKerbinOrbit(kerbolFrame)
  val kerbinFrame = createKerbinFrame(kerbolFrame, kerbinOrbit)
  val kerbin = CustomCelestialBody("Kerbin", KerbolBodyConstants.KERBIN_MU, kerbinFrame, kerbinOrbit)

  // Create the orbit for Mun
  val munOrbit = createMunOrbit(kerbinFrame)
  val munFrame = createMunFrame(kerbinFrame, munOrbit)
  val mun = CustomCelestialBody("Mun", KerbolBodyConstants.MUN_MU, munFrame, munOrbit)

}